NuRadioMC/simulation/simulation.py

```diff
@@ -401,752 +1339,325 @@ def __init__(self, inputfilename,
-            self._eventWriter.begin(self._outputfilenameNuRadioReco, log_level=self._log_level)
-        unique_event_group_ids = np.unique(self._fin['event_group_ids'])
-        self._n_showers = len(self._fin['event_group_ids'])
-        self._shower_ids = np.array(self._fin['shower_ids'])
-        self._shower_index_array = {}  # this array allows to convert the shower id to an index that starts from 0 to be used to access the arrays in the hdf5 file.
-
-        self._raytracer = self._prop(
-            self._ice, self._cfg['propagation']['attenuation_model'],
-            log_level=self._log_level_ray_propagation,
-            n_frequencies_integration=int(self._cfg['propagation']['n_freq']),
-            n_reflections=self._n_reflections,
-            config=self._cfg,
-            detector=self._det
-        )
-        for shower_index, shower_id in enumerate(self._shower_ids):
-            self._shower_index_array[shower_id] = shower_index
+        logger.status("Starting NuRadioMC simulation")
+        self.__time_logger.reset_times()
 
-        self._create_meta_output_datastructures()
-
-        # check if the same detector was simulated before (then we can save the ray tracing part)
-        pre_simulated = self._check_if_was_pre_simulated()
-
-        # Check if vertex_times exists:
-        self._check_vertex_times()
-
-        input_time = 0.0
-        askaryan_time = 0.0
-        rayTracingTime = 0.0
-        detSimTime = 0.0
-        outputTime = 0.0
-        weightTime = 0.0
-        distance_cut_time = 0.0
+        efieldToVoltageConverter.begin(time_resolution=self._config['speedup']['time_res_efieldconverter'])
+        channelGenericNoiseAdder.begin(seed=self._config['seed'])
+        if self._outputfilenameNuRadioReco is not None:
+            eventWriter.begin(self._outputfilenameNuRadioReco, log_level=self._log_level)
 
-        n_shower_station = len(self._station_ids) * self._n_showers
-        iCounter = 0
+        particle_mode = "simulation_mode" not in self._fin_attrs or self._fin_attrs['simulation_mode'] != "emitter"
+        event_group_ids = np.array(self._fin['event_group_ids'])
+        unique_event_group_ids = np.unique(event_group_ids)
 
         # calculate bary centers of station
-        self._station_barycenter = np.zeros((len(self._station_ids), 3))
+        station_barycenter = np.zeros((len(self._station_ids), 3))
         for iSt, station_id in enumerate(self._station_ids):
             pos = []
-            for channel_id in range(self._det.get_number_of_channels(station_id)):
+            for channel_id in self._det.get_channel_ids(station_id):
                 pos.append(self._det.get_relative_position(station_id, channel_id))
-            self._station_barycenter[iSt] = np.mean(np.array(pos), axis=0) + self._det.get_absolute_position(station_id)
+            station_barycenter[iSt] = np.mean(np.array(pos), axis=0) + self._det.get_absolute_position(station_id)
```

Comment:
comprehensive lists?

Reply:
what do you mean?

Reply:
You can compress the code into:
```
        station_barycenter = np.array([
            np.mean([self._det.get_relative_position(station_id, channel_id) for channel_id in self._det.get_channel_ids(station_id)], axis=0) + self._det.get_absolute_position(station_id)
            for station_id in self._station_ids])
```
or 
```
        # calculate bary centers of station
        station_barycenter = np.zeros((len(self._station_ids), 3))
        for iSt, station_id in enumerate(self._station_ids):
            station_barycenter[iSt] = np.mean([self._det.get_relative_position(station_id, channel_id) for channel_id in self._det.get_channel_ids(station_id)], axis=0) + self._det.get_absolute_position(station_id)
```

Reply:
And, cant we move this calculation into the init function?